explorer/interpreter/interpreter.cpp

```diff
@@ -1449,15 +1449,42 @@ auto Interpreter::StepExp() -> ErrorOr<Success> {
           CARBON_CHECK(args.size() == 1);
           CARBON_CHECK(act.pos() > 0);
           const auto* ptr = cast<PointerValue>(args[0]);
-          if (act.pos() == 1) {
-            CARBON_ASSIGN_OR_RETURN(
-                const auto* pointee,
-                this->heap_.Read(ptr->address(), exp.source_loc()));
-            return todo_.Spawn(std::make_unique<DestroyAction>(
-                arena_->New<LValue>(ptr->address()), pointee));
+          CARBON_ASSIGN_OR_RETURN(const auto* pointee,
+                                  heap_.Read(ptr->address(), exp.source_loc()));
+          if (const auto* class_value = dyn_cast<NominalClassValue>(pointee)) {
+            // Handle destruction from base class pointer.
+            const auto* child_class_value = *class_value->class_value_ptr();
```

Comment:
Might actually not need it anymore if we rely on the `Address`, as we can get the value back from the Address directly.